encoding/protoavro/encode.go

```diff
@@ -47,69 +47,72 @@ func messageJSON(message protoreflect.Message) (interface{}, error) {
 			continue
 		}
 		value := message.Get(field)
-		jsonValue, err := fieldJSON(field, value)
+		jsonValue, err := o.fieldJSON(field, value)
 		if err != nil {
 			return nil, err
 		}
 		record[string(field.Name())] = jsonValue
 	}
+	if o.OmitRootElement && message.Descriptor().Index() == 0 {
```

Comment:
Why is `.Descriptor().Index` necessary here?

Reply:
Was removed in later commits